aiocache/base.py

```diff
@@ -109,18 +110,26 @@ class BaseCache:
     NAME: str
 
     def __init__(
-        self, serializer=None, plugins=None, namespace=None, key_builder=None, timeout=5, ttl=None
+        self,
+        serializer: Optional[BaseSerializer] = None,
+        plugins: Optional[List[Any]] = None,  # aiocache.plugins depends on aiocache.base
+        namespace: str = "",
+        key_builder: Optional[Callable[[str, str], str]] = None,
+        timeout: Optional[float] = 5,
+        ttl: Optional[float] = None,
     ):
-        self.timeout = float(timeout) if timeout is not None else timeout
-        self.namespace = namespace
-        self.ttl = float(ttl) if ttl is not None else ttl
-        self.build_key = key_builder or self._build_key
+        self.timeout: Optional[float] = float(timeout) if timeout is not None else None
+        self.ttl: Optional[float] = float(ttl) if ttl is not None else None
 
-        self._serializer = None
-        self.serializer = serializer or serializers.StringSerializer()
+        self.namespace: str = namespace
+        self._build_key: Callable[[str, str], str] = key_builder or (
+            lambda key, namespace="": f"{namespace}{key}")
 
-        self._plugins = None
-        self.plugins = plugins or []
+        self._serializer: Optional[BaseSerializer] = None
+        self.serializer: BaseSerializer = serializer or StringSerializer()
```

Comment:
Here and elsewhere, should we replace the base class with a `TypeVar`?

Something like...
```
from typing import TypeVar
SerializerType = TypeVar('SerializerType', bound='BaseSerializer')

self.serializer: SerializerType = serializer or StringSerializer()
```

Reply:
I'm not sure what benefit that brings. But, leave it to a later PR if you think there is a benefit.

Reply:
That's to indicate that the type can be a subclass of the bound type; so `StringSerializer` is not expected to be coerced to `BaseSerializer`, for example.

> But, leave it to a later PR if you think there is a benefit.

Agreed.